src/bun.js/node/node_fs.zig

```diff
@@ -5119,6 +5129,8 @@ pub const NodeFS = struct {
         comptime is_root: bool,
     ) Maybe(void) {
         const flags = os.O.DIRECTORY | os.O.RDONLY;
+        const dirent_path = bun.String.createUTF8(basename);
```

Comment:
Do we need to create the `dirent_path` if `ExpectedType` != `Dirent`?

Is it possible to reuse the input path the user passed instead of converting to UTF-8, then creating a duplicate copy of that?